app/src/composables/use-mutex.ts

```diff
@@ -0,0 +1,61 @@
+import { sleep } from '@/utils/sleep';
+
+const TIMEOUT = 500;
+const MAX_RETRIES = 10;
+
+export function useMutex(key: string, expiresMs: number) {
+	const internalKey = `directus-mutex-${key}`;
+	const useWebLock = !!navigator.locks;
+
+	async function acquireMutex(callback: (lock?: Lock | null) => Promise<any>): Promise<any> {
+		if (useWebLock) {
+			const controller = new AbortController();
+
+			// abort if a lock is not acquired or is not released within the expiry time
+			setTimeout(() => controller.abort(), expiresMs);
```

Comment:
Does this make sense? this setTimeout covers both "no lock acquired within the expiry" and "the lock not released within expiry" 🤔 Could there be a case where acquiring the lock takes very long but still succeeds and then incorrectly gets aborted?

We do have the expiry time set to 5seconds so seems unlikely but yet a refresh call could theoretically take longer than that without erroring.

Reply:
IMO it does make sense, since is mostly a precaution against blocking other windows/tabs from never returning out of the `refresh` method. 

In the worst case case I can come up with here (one tab takes longer than 5s to return from the refresh call), the other tabs will wait for 5s and then continue without having called refresh. If the token is expired, they will be logged out automatically and either have to log in again or refresh the page once the initial tab comes back from the refresh call and the correct cookie has been set.